### DISCUSSION TITLE
what is this watermelon?

### DISCUSSION THREAD

**Discussion by jovlinger** [zed-industries/zed#12227] [Feature Requests]
I was coding and then I got a watermelon. Why. What is it. Make it go away

![image](https://github.com/zed-industries/zed/assets/10566845/e25ba7f2-7c28-4bde-9d3e-0056db11fcfc)

**Comment by jovlinger** [comment]
(to be clear, escape makes it go away, so that part is hyperbole, but the rest is just surprised pikachu)

**Comment by SomeoneToIgnore** [comment]
This is a git hunk diff, that's supposed to appear either on the gutter click, or on toggling those via `editor: toggle hunk diff`.
Could be so, that you've clicked on the gutter or pressed `cmd-'` and that appeared.

We have a logic to automatically re-expand hunks if those were expanded before, the buffer contents changed and the change was adjacent to the old expansion — that is not supposed to happen when no hunks are expanded, but if it does, would be nice to get the reproduction steps from your side.

**Reply by jovlinger** [reply]
Thank you for a clear answer!

**Reply by jovlinger** [reply]
I think I was aiming for another key combo and missed. Good explanation.
